src/test/util/mod.rs

```diff
@@ -256,14 +257,15 @@ impl TestClient {
         drop_collection(&coll).await;
     }
 
-    pub fn topology(&self) -> String {
+    pub fn topology(&self) -> Topology {
         if self.is_sharded() {
-            String::from("sharded")
+            Topology::Sharded
         } else if self.is_replica_set() {
-            String::from("replicaset")
+            Topology::ReplicaSet
         } else {
-            String::from("single")
+            Topology::Single
         }
+        // TODO parse out sharded replica set
```

Comment:
In 3.6+, shards must be replica sets, and since we only support 3.6+ perhaps we only need the single `Sharded` case.

Reply:
Are you sure about that? I'm pretty sure I've made a sharded cluster with singleton shards with a recent server version.

Reply:
To be honest, I thought I had too, but according to the docs you can't do that in 3.6: https://docs.mongodb.com/manual/core/sharded-cluster-components/. It's possible the docs are incorrect though.

Reply:
I just tried it with 4.4, and it works fine. I guess the docs are wrong.